keystores/src/main/java/com/palantir/conjure/java/config/ssl/KeyStores.java

```diff
@@ -319,8 +328,9 @@ private static KeyStore createKeyStore() {
     }
 
     private static List<Certificate> readX509Certificates(InputStream certificateIn) throws CertificateException {
-        CertificateFactory certFactory = CertificateFactory.getInstance("X.509");
-        return new ArrayList<>(certFactory.generateCertificates(certificateIn));
+        return CertificateFactory.getInstance("X.509").generateCertificates(certificateIn).stream()
```

Comment:
The difference is here. `CertificateFactory#generateCertificates` for X509 doesn't deduplicate certificates globally  while `CertificateFactory#generateCertificate` does